src/main/java/com/coveo/spillway/memory/InMemoryStorage.java

```diff
@@ -0,0 +1,71 @@
+package com.coveo.spillway.memory;
+
+import com.coveo.spillway.AddAndGetRequest;
+import com.coveo.spillway.LimitKey;
+import com.coveo.spillway.LimitUsageStorage;
+
+import java.time.Instant;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+public class InMemoryStorage implements LimitUsageStorage {
+
+  Map<Instant, Map<LimitKey, AtomicInteger>> map = new ConcurrentHashMap<>();
+  private Object lock = new Object();
+
+  @Override
+  public Map<LimitKey, Integer> addAndGet(Collection<AddAndGetRequest> requests) {
+    Map<LimitKey, Integer> updatedEntries = new HashMap<>();
+    synchronized (lock) {
+      for (AddAndGetRequest request : requests) {
+        Instant expirationDate = request.getBucket().plus(request.getExpiration());
+
+        LimitKey limitKey = LimitKey.fromRequest(request);
+
+        Map<LimitKey, AtomicInteger> mapWithThisExpiration =
+            map.computeIfAbsent(expirationDate, (key) -> new HashMap<>());
+        AtomicInteger counter =
+            mapWithThisExpiration.computeIfAbsent(limitKey, (key) -> new AtomicInteger(0));
+        updatedEntries.put(limitKey, counter.addAndGet(request.getCost()));
+      }
+      removeExpiredEntries();
```

Comment:
Ok. Probably doesn't take a long time anyway. I was asking because the call to removeExpiredEntries is not synchronize elsewhere and maybe the call to addAndGet could block others while doing the remove.